### PR TITLE
Build Python 3.13 wheels, drop support for Python 3.7

### PR DESCRIPTION
**Pull request by edgarrmondragon**
I found it quite difficult to continue building 3.7 wheels for macos. Maybe someone knows better.

See https://github.com/psycopg/psycopg2/issues/1692.

### CODE REVIEW

File: doc/src/install.rst [hunk 1]
```diff
@@ -131,7 +131,7 @@ The current `!psycopg2` implementation supports:
 ..
     NOTE: keep consistent with setup.py and the /features/ page.
 
-- Python versions from 3.7 to 3.12
+- Python versions from 3.8 to 3.12
```
Comment (dvarrazzo) [inline doc/src/install.rst:134]: This can go to 3.13

**Comment by dvarrazzo** [comment]
Trying to [build the packages](https://github.com/psycopg/psycopg2/actions/runs/11232596331)... :eyes:

File: doc/src/install.rst (hunk 1 above)
Comment (dvarrazzo) [reply doc/src/install.rst]: Done it myself

**Comment by dvarrazzo** [comment]
Looking into the build failure on appveyor. It seems like the database used for tests is too old, trying to use a more recent one.

**Comment by edgarrmondragon** [comment]
@dvarrazzo are there any more blockers?

**Comment by dvarrazzo** [comment]
Yesterday I managed to run the test, but not the packaging. Trying that now.

**Comment by dvarrazzo** [comment]
I managed to build packages on Windows too. We need to drop Python 3.8 but it doesn't seem a problem.

There are a couple of little bugs I'd like to fix:

- #1728 to re-test and merge
- #1724 to fix, but it should be easy

So I'd like to get those in and make a release.

**Pull request merged**